### ISSUE TITLE
QueueDoesNotExistException from Lambda on 0.14.3

### ISSUE THREAD

**Issue by xtaixe** [localstack/localstack#6176]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have a [reproducer](https://github.com/localstack/localstack/files/8805292/reproducer.zip) with a Lambda that has a queue as an event source and publishes to another queue. It works on `0.14.2` but on `0.14.3` it gives he following error when trying to publish to SQS from the Lambda: 

```
The specified queue does not exist for this wsdl version. (Service: AmazonSQS; Status Code: 400; Error Code: AWS.SimpleQueueService.NonExistentQueue; Request ID: 5KOM5KINTCY7CBJVGYXB2SLYZ978GSR1VIJNXM0E01X8TX0CLZDL; Proxy: null): com.amazonaws.services.sqs.model.QueueDoesNotExistException
reproducer-localstack-1  | com.amazonaws.services.sqs.model.QueueDoesNotExistException: The specified queue does not exist for this wsdl version. (Service: AmazonSQS; Status Code: 400; Error Code: AWS.SimpleQueueService.NonExistentQueue; Request ID: 5KOM5KINTCY7CBJVGYXB2SLYZ978GSR1VIJNXM0E01X8TX0CLZDL; Proxy: null)
reproducer-localstack-1  | 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1879)
```

### Expected Behavior

The Lambda publishes to SQS and the message can be retrieved succesfully, as in version `0.14.2`.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack

Unzip the [reproducer](https://github.com/localstack/localstack/files/8805292/reproducer.zip) and run `docker compose up`, which will already create the queues and the lambda.

#### Client commands

`aws --region ap-south-1 --endpoint-url http://localhost:4566 sqs send-message --queue-url "http://localstack:4566/000000000000/dx-ref-java-quarkus-im-lambda-sqs-in" --message-body "hello"`

to trigger the lambda.

On `0.14.2` you can then retrieve the message with:
`aws --region ap-south-1 --endpoint-url http://localhost:4566 sqs receive-message --queue-url "http://localstack:4566/000000000000/dx-ref-java-quarkus-im-lambda-sqs-out"`

### Environment

```markdown
- OS: macOS 12.4
- LocalStack: 0.14.3
```

### Anything else?

The Lambda basically does this:

```java
    public Void handleRequest(SQSEvent event, Context context) {
        sendResult(event.getRecords().get(0).getBody());

        return null;
    }

    void sendResult(String result) {
        // Get the result queue name from env var and the necessary data from it's own env vars (built from the name)
        final var resultQueueName = getNonEmptyEnvVar("OUT_QUEUE_NAME");
        final var queueUrl = getNonEmptyEnvVar(getEnvVarName("SQS_%s_QUEUE_URL", resultQueueName));
        final var queueRegion = getNonEmptyEnvVar(getEnvVarName("SQS_%s_QUEUE_REGION", resultQueueName));
        final var queueEndpoint = getEnvVar(getEnvVarName("SQS_%s_QUEUE_ENDPOINT", resultQueueName));

        System.out.println("OUT_QUEUE_NAME: " + resultQueueName);
        System.out.println("QUEUE_URL: " + queueUrl);
        System.out.println("QUEUE_REGION: " + queueRegion);
        System.out.println("QUEUE_ENDPOINT: " + queueEndpoint);

        final var sqsClient = getSqsClient(queueEndpoint, queueRegion);

        sqsClient.sendMessage(queueUrl, result);
    }

    private AmazonSQS getSqsClient(String endpoint, String region) {
        return endpoint == null || endpoint.isBlank()
                ? AmazonSQSClient.builder().withRegion(region).build()
                : AmazonSQSClient.builder().withEndpointConfiguration(new EndpointConfiguration(endpoint, region)).build();
    }

    private String getEnvVarName(String envVarPattern, String resultQueueName) {
        return format(envVarPattern, resultQueueName.replace("-", "_").toUpperCase());
    }

    String getEnvVar(String envVar) {
        return System.getenv(envVar);
    }

    String getNonEmptyEnvVar(String envVar) {
        final var value = getEnvVar(envVar);

        if (value == null || value.isBlank()) {
            throw new IllegalStateException(format("%s env var is empty", envVar));
        }

        return value;
    }
```

**Comment by CarlosLanderas** [comment]
I have the same problem with recorded_api calls. When I create a container with a volume that has recorded API calls with version 0.14.3 the SQS definitions are simply not saved.

When I downgrade to version 0.14.2 they start working back again.

**Comment by xtaixe** [comment]
Hi. Since this has a reproducer, can we expect it to be triaged/fixed soon to set expectations on our side or what's the usual process? Thanks. cc @dfangl

**Comment by thrau** [comment]
Hi @xtaixe, sorry for the delay! I finally got some time to debug the issue and traced it to the same issue we had in #6102, which has since been fixed with #6145.

I was no longer able to reproduce the issue with 1.0.3 (should also be fixed in 0.14.4). One way to keep using 0.14.3 would be to set `SQS_ENDPOINT_STRATEGY=path`

Basically our implementation of the SQS Query API would in 0.14.3 not resolve the region automatically if you used the `SQS_ENDPOINT_STRATEGY=path`. With the fix in #6145, we reverted that decision and now you can see in the logs a line:
```
localstack_1  | 2022-08-04T14:20:12.313 DEBUG --- [   asgi_gw_0] l.services.sqs.query_api   : Region of queue URL http://localstack:4566/000000000000/dx-ref-java-quarkus-im-lambda-sqs-out is ambiguous, got region ap-south-1 from request
```

if you use `SQS_ENDPOINT_STRATEGY=path`, then the region is encoded into the path of the queue url. [here is the section in our docs](https://docs.localstack.cloud/aws/sqs/#queue-urls).

and, thanks for the great reproduction example! please re-open a new issue if the problem resurfaces.
